packages/apollo-forest-run/src/forest/types.ts

```diff
@@ -44,6 +44,9 @@ export type IndexedTree = {
 
   // ApolloCompat
   danglingReferences?: Set<NodeKey>;
+
+  // Metadata about chunk updates, used for telemetry and debugging
+  chunksMetadata: UpdateChunkMetadata[];
```

Comment:
After some consideration I now think we should not put update details to IndexedTree. Let's make `updateTree` return `UpdateState` and include update stats + indexedTree in this state object. It is already created today, so the overhead should be minimal.